File: src/main/java/de/sean/splugin/util/SUtil.java
package de.sean.splugin.util;

import org.bukkit.Material;
import org.bukkit.OfflinePlayer;
import org.bukkit.entity.Player;
import org.bukkit.inventory.ItemStack;
import org.bukkit.inventory.meta.ItemMeta;
import org.bukkit.inventory.meta.SkullMeta;
import org.jetbrains.annotations.NotNull;

import java.awt.*;
import java.util.List;
import java.util.*;
import java.util.concurrent.ThreadLocalRandom;

public class SUtil {
    private static final HashMap<UUID, Long> playerLastActivity = new HashMap<>();
    private static final HashMap<UUID, Boolean> afkPlayers = new HashMap<>();

    public static void removePlayerAFK(UUID uuid)                       {           afkPlayers.remove(uuid); }
    public static void setPlayerAFK(UUID uuid, boolean afk)             {           afkPlayers.put(uuid, afk); }
    public static boolean isPlayerAFK(UUID uuid)                        { return    afkPlayers.get(uuid); }

    public static void removeActivityForPlayer(UUID uuid)               {           playerLastActivity.remove(uuid); }
    public static void setLastActivityForPlayer(UUID uuid, long time)   {           playerLastActivity.put(uuid, time); }
    public static long getLastActivityForPlayer(UUID uuid)              { return    playerLastActivity.get(uuid); }

    @NotNull
    public static String concatArrayRange(String[] arr, int begin, int end) {
        final StringBuilder builder = new StringBuilder();
        for (int i = begin; i < end; i++) {
            builder.append(arr[i]).append(" ");
        }
        return builder.toString();
    }

    public static List<String> parseStringList(String str) {
        final List<String> ret = new ArrayList<>(Arrays.asList(str.replaceAll("^\\[|]$", "").split(",")));
        ret.removeIf(String::isEmpty);
        return ret;
    }

    /**
     * Returns a pseudorandom {@code int} value between the specified
     * origin (inclusive) and the specified bound (exclusive).
     *
     * @param min the least value returned
     * @param max the upper bound (exclusive)
     * @return a pseudorandom {@code int} value between the origin
     *         (inclusive) and the bound (exclusive)
     * @throws IllegalArgumentException if {@code origin} is greater than
     *         or equal to {@code bound}
     */
    public static int randomInt(int min, int max) throws IllegalArgumentException {
        return ThreadLocalRandom.current().nextInt(min, max);
    }

    public static Color randomColor() {
        final Color[] colors = {Color.BLACK, Color.BLUE, Color.CYAN, Color.DARK_GRAY, Color.GRAY, Color.LIGHT_GRAY, Color.GREEN, Color.MAGENTA, Color.ORANGE, Color.PINK, Color.RED, Color.WHITE, Color.YELLOW};
        return colors[randomInt(0, colors.length)];
    }

    public static ItemStack getItemStack(int num, Material material, String name) {
        return getItemStack(num, material, name, Collections.emptyList());
    }

    public static ItemStack getItemStack(int num, Material material, String name, List<String> lore) {
        ItemStack stack = new ItemStack(material, num);
        ItemMeta meta = stack.getItemMeta();
        meta.setDisplayName(name);
        meta.setLore(lore);
        stack.setItemMeta(meta);
        return stack;
    }

    public static ItemStack getPlayerSkull(OfflinePlayer player) {
        ItemStack skull = new ItemStack(Material.PLAYER_HEAD, 1);
        SkullMeta meta = (SkullMeta) skull.getItemMeta();
        meta.setOwningPlayer(player);
        meta.setDisplayName(player.getName());
        skull.setItemMeta(meta);
        return skull;
    }

    public static ItemStack getPlayerSkull(Player player) {
        ItemStack skull = new ItemStack(Material.PLAYER_HEAD, 1);
        SkullMeta meta = (SkullMeta) skull.getItemMeta();
        meta.setOwningPlayer(player);
        meta.setDisplayName(player.getDisplayName());
        skull.setItemMeta(meta);
        return skull;
    }
}

File: src/main/kotlin/de/sean/splugin/util/SUtil.kt
package de.sean.splugin.util

import org.bukkit.Material
import org.bukkit.OfflinePlayer
import org.bukkit.entity.Player
import org.bukkit.inventory.ItemStack
import org.bukkit.inventory.meta.SkullMeta
import java.awt.Color
import java.util.*
import java.util.concurrent.ThreadLocalRandom
import kotlin.collections.HashMap

object SUtil {
    val playerLastActivity = HashMap<UUID, Long>()
    val afkPlayers = HashMap<UUID, Boolean>()
    val colors = arrayOf(Color.BLACK, Color.BLUE, Color.CYAN, Color.DARK_GRAY, Color.GRAY, Color.LIGHT_GRAY, Color.GREEN, Color.MAGENTA, Color.ORANGE, Color.PINK, Color.RED, Color.WHITE, Color.YELLOW)

    fun concatArrayRange(arr: Array<String>, begin: Int, end: Int): String {
        val builder = StringBuilder()
        for (i in begin until end) {
            builder.append(arr[i]).append(" ")
        }
        return builder.toString()
    }

    fun parseStringList(str: String): List<String> {
        val ret: MutableList<String> = ArrayList(listOf(*str.replace("^\\[|]$".toRegex(), "").split(",").toTypedArray()))
        ret.removeIf { obj: String -> obj.isEmpty() }
        return ret
    }

    /**
     * Returns a pseudorandom `int` value between the specified
     * origin (inclusive) and the specified bound (exclusive).
     *
     * @param min the least value returned
     * @param max the upper bound (exclusive)
     * @return a pseudorandom `int` value between the origin
     * (inclusive) and the bound (exclusive)
     * @throws IllegalArgumentException if `origin` is greater than
     * or equal to `bound`
     */
    @Throws(IllegalArgumentException::class)
    fun randomInt(min: Int, max: Int): Int {
        return ThreadLocalRandom.current().nextInt(min, max)
    }

    fun randomColor(): Color {
        return colors[randomInt(0, colors.size)]
    }

    fun getItemStack(num: Int, material: Material?, name: String?): ItemStack {
        return getItemStack(num, material, name, emptyList<String>())
    }

    fun getItemStack(num: Int, material: Material?, name: String?, lore: List<String?>?): ItemStack {
        val stack = ItemStack(material!!, num)
        val meta = stack.itemMeta
        meta!!.setDisplayName(name)
        meta.lore = lore
        stack.itemMeta = meta
        return stack
    }

    fun getPlayerSkull(player: OfflinePlayer): ItemStack {
        val skull = ItemStack(Material.PLAYER_HEAD, 1)
        val meta = skull.itemMeta as SkullMeta?
        meta!!.owningPlayer = player
        meta.setDisplayName(player.name)
        skull.itemMeta = meta
        return skull
    }

    fun getPlayerSkull(player: Player): ItemStack {
        val skull = ItemStack(Material.PLAYER_HEAD, 1)
        val meta = skull.itemMeta as SkullMeta?
        meta!!.owningPlayer = player
        meta.setDisplayName(player.displayName)
        skull.itemMeta = meta
        return skull
    }
}
